src/language/ast.ts

```diff
@@ -63,15 +82,99 @@ export interface ASTKindToNode {
   ModelDescription: ModelDescriptionNode;
   Name: NameNode;
   OptionalName: OptionalNameNode;
+  // Endpoints types
+  EndpointNamespaceTypeDefinition: EndpointNamespaceTypeDefinitionNode;
+  EndpointVerb: EndpointVerbNode;
+  EndpointUrl: EndpointUrlNode;
+  EndpointTypeDefinition: EndpointTypeDefinitionNode;
+  EndpointResponse: EndpointResponseNode;
+  EndpointSecurityDefinition: EndpointSecurityDefinitionNode;
 }
 
 export interface DocumentNode {
   readonly kind: 'Document';
   readonly loc?: Location;
-  readonly definitions: ReadonlyArray<TypeDefinitionNode>;
+  readonly definitions: ReadonlyArray<TypeDefinitionNode | EndpointNamespaceTypeDefinitionNode>;
+}
+
+export type TypeDefinitionNode =
+  | ModelTypeDefinitionNode
+  | EnumTypeDefinitionNode
+
+export interface EndpointNamespaceTypeDefinitionNode {
+  readonly kind: 'EndpointNamespaceTypeDefinition';
+  readonly tag?: NameNode;
+  readonly description?: ModelDescriptionNode;
+  readonly endpoints?: ReadonlyArray<EndpointTypeDefinitionNode>;
+  readonly loc?: Location;
+}
+
+export interface EndpointVerbNode {
+  readonly kind: 'EndpointVerb';
```

Comment:
if we already do concatenate of namespace in ast what about to do next thing?

https://www.typescriptlang.org/play?#code/C4TwDgpgBAggygFQMIHsB2wIA9gFE0AmYKAlhlALxQDk+RpG1UAUKJLIqhtsALIoEIAG0pQa-QUOrMZ2YgCdgUMpnkAzAIYBjaHWIqAahHkAjAHIDoAb2Zj5EDQXRCQUANZkCALigADACRW8MjomDh6DMAAvkamvgDcMlFAA